dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Account/Email/Resend.cshtml.cs

```diff
@@ -0,0 +1,73 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using TeacherIdentity.AuthServer.Models;
+using TeacherIdentity.AuthServer.Pages.Common;
+using TeacherIdentity.AuthServer.Services.UserVerification;
+
+namespace TeacherIdentity.AuthServer.Pages.Account.Email;
+
+public class Resend : BaseEmailPageModel
+{
+    private ProtectedStringFactory _protectedStringFactory;
+
+    public Resend(
+        IUserVerificationService userVerificationService,
+        IIdentityLinkGenerator linkGenerator,
+        TeacherIdentityServerDbContext dbContext,
+        ProtectedStringFactory protectedStringFactory) :
+        base(userVerificationService, linkGenerator, dbContext)
+    {
+        _protectedStringFactory = protectedStringFactory;
+    }
+
+    [BindProperty]
+    [Display(Name = "Email address")]
+    [Required(ErrorMessage = "Enter your new email address")]
+    [EmailAddress(ErrorMessage = "Enter a valid email address")]
+    public string? NewEmail { get; set; }
```

Comment:
Naming this Email caused it to automatically bind to the GET query parameter, so had to slightly change the signature of GenerateEmailPinForNewEmail